db/config.go

```diff
@@ -0,0 +1,14 @@
+package db
+
+import (
+	"time"
+)
+
+type Config struct {
+	Driver              string        `yaml:"driver" mapstructure:"driver"`
```

Comment:
do we need these `yaml` & `mapstructure` tags here?

Reply:
So that when we compose this in a higher level config struct, we can tell viper what names to use in the YAML-based config. Any other suggestion to achieve that?

Reply:
I think this the best way to do it. 

P.S.: I think, these kinds of tags are okay (even good) to be there for any struct even if you don't have immediate need -- They don't really have an inherent cost and when you actually need to use the struct to show/read yaml/json etc. it just works. (It just says "if you were to represent this in YAML, these are the key names you should be using")

Reply:
+1